File: loops/notes.c
// Joslynn Ramirez, Loop notes c
#include <stdio.h>
#include <string.h>
int main(void){
// 1. What is a loop?
    //a section of code that gets repeated multiple times

// 2. What are the two types of loops
    //While loops
    int start = 0;
    while(start <5){
        printf("Hello!\n");
        start++;
    }
    //for loop
    int q;
    for(q=0;q<5;q++){
        printf("%d\n", q);
    }
// 3. What is iteration
    // Repeating something with minor changes 

// 4. What are array (list)?
    // a variable that holds multiple values

// 8. How do you make array in c?
    // arrays all need to be the same data type
int grades[] = {88, 97, 100, 70, 72, 99, 61};
   // 1. set data type 2. AFTER naming put brackets and write the length of the list 3. list is surrounded by curly brackets {} 4. Commas , between each item
// print a single item
printf("CSP grade: %d\n", grades[2]);
//change an item in the array
grades[2] = 95;
printf("CSP grade: %d\n", grades[2]);
//size of list in bytes
int size = sizeof(grades);
// length in list items
int length = sizeof(grades)/sizeof(grades[0]);
printf("%d\n", length);
// array with strings
// First brackets set the length of the list
//seconds brackets sets the length of each string 
char movies[][20] = {"Cars", "Trausre Planet", "An American Tale", "Marley and m=Me", "The Avengers"};
printf("The movie is %s!\n", movies[1]);
int mlength = sizeof(movies)/sizeof(movies[0])
// 9. How do you make loops for C

// set the iterator, keeps track of how many times ive gone through the loop (best practice to set at x or i)
int x;
// in parenthesis (Starting point; ending point; what we count by)
for(x = 0; x <= 10; x+=2){
    printf("%d\n", x);
}
int m;
for(m=0;m<mlength;m++){
    printf("%s\n", movies[m]);
}
// 10. How do you make while loops in C
int w = 0;

while(w<100){
    printf("%d\n", w);
    w += 10;
}
    return 0;
}

File: loops/notes.py
#Joslynn Ramirez, Loops Notes python

#What is a loop
   #section of code that repeats multiple times

#2. What are the two types of loops
     #For loop
nums = [12, 3, 66, 7, 3, 3, 2]

for num in nums:
    print(num)

#While loop
x = 0

while x < 10:
    print(x)
    x+= 1
#3. What is iteration
    #That specific instance of the loop
    #Iteration the amount of times you are looping through
    
#4. What are lists
    # A variable that holds multiple values
nums = [1,2,3,4,5,6,7] #No ""
friends = ["Yenesis", "David", "Johann", "Juan"] #""
print(nums) #prints whole list is ugly
print(friends[3]) #prints the 3rd one

friends[4] = "Juanito"
friends. pop(2)
friends.insert(1, "Evelyn")
friends.insert(2, ["person", "person_two", "person_three"])
print(friends)



#5. How do you make lists in Python
    #steps to make a list
    #step 1. Use []
    #add in items with correct data types
#Have a coma between each items

#6. How do you make loops in Python?
for friends in friends:
    print(friends)

for x in range(1,5):
    print(x)

#7. How do you make while loops in Python?
import random 
x = 1 #Keeps count of the iteration
goose = random. randint(1,20)

while x <= 20:
    if x == goose:
        print("Goose!")
        break #Tells the loops to stop
    else:
        print("Duck")
    x+= 1

#Continue moves on to the next iteration without finishing